### DISCUSSION TITLE
Use message from WebSockets for graceful shutdown

### DISCUSSION THREAD

**Discussion by ArijitSinghEDA** [fastapi/fastapi#11199] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, BackgroundTasks, WebSocket
from pydantic import BaseModel
from main_process import main

class item(BaseModel):
    params: dict

app = FastAPI()

@app.post("/")
async def func(item: Item, background_tasks: BackgroundTasks):
     background_tasks.add_task(main, item.params)

@app.websocket("/ws")
async def web_socket(websocket: WebSocket):
    # need help here
```

### Description

I have a file structure in this way

```
Main Folder
|--main.py
|--main_process.py
|--gunicorn_config.py
```

I am running the following

```
gunicorn main:app -c gunicorn_config.py
```

Now I wish to get a message from the `WebSocket`, and use that message to gracefully shutdown my running process. How do I achieve this?

### Operating System

Linux

### Operating System Details

WSL with Ubuntu 22.04

### FastAPI Version

0.103.1

### Pydantic Version

1.10.12

### Python Version

3.10.12

### Additional Context

Any alternatives are appreciated too. My main aim is to get a message from the user (via a click of a button), and then gracefully shutdown the process for the specific user (I have access to the custom ID of each process a user is running).

**Comment by jgould22** [comment]
You can probably use https://docs.python.org/3/library/signal.html and https://docs.gunicorn.org/en/stable/signals.html#master-process

**Reply by ArijitSinghEDA** [reply]
Wouldn't this kill/interrupt the worker? I just want to terminate a single process running on any worker.

**Reply by jgould22** [reply]
Process is probably the wrong word to use then as it has a very specific meaning in this context. A worker is a single process, it runs tasks or co-routines. 

Can you elaborate on what you mean by process in this context.

Do you mean the current request being handled by the `/ws` endpoint ?

Do you mean a background task spawned by your `/` endpoint ?

**Reply by ArijitSinghEDA** [reply]
Thank you for the clarification.

By process, I mean the task that I am sending in the background_tasks by my `/` endpoint.

Currently I am not using the `/ws` endpoint. I saw it in the documentation, and wanted to apply it in my use case.

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
